server/routes/profile.js

```diff
@@ -0,0 +1,38 @@
+const express = require("express");
+const mongoose = require("mongoose");
+const User = require("../models/User");
+const passport = require("passport");
+
+const router = express.Router();
+
+router.patch("/",
+  passport.authenticate('jwt', { session: false }),
+  async (req, res) => {
+    try {
+      const id = req.user._id;
+      const user = await User.findById(id);
+      
+      if (user) {
+        const {
+          name,
+          profileUrl,
+        } = req.body;
+        if (name) user.name = name;
+        if (profileUrl) user.profileUrl = profileUrl;
+
+        const body = await user.save();
```

Comment:
you can add a then() block to this, and send back the user object instead of doing body